### PR TITLE
docs: add examples for GridPro cell editability

### LINKED ISSUE #3355: Add docs for Grid Pro conditional cell editability
In the `next` branch of docs:
Add new top-level section above _Distinguish Editable / Read-Only Cells_:

> ### Conditional Editability `Since 24.4`
> In some situations you might need to disable editing of specific cells in an edit column. This can be achieved through a function that determines the editability of each cell. This function is automatically re-evaluated when an item in the Grid is modified, so that modifying the value in one cell can immediately affect the editability of other cells.
> 
> `[sample: the example used in the R&D demo by Sascha was fine, so we could probably just use that]`
>

### PR DESCRIPTION
**Pull request by web-padawan**
Fixes #3355

Added the example based on the Flow components IT for the feature:

Screenshot 2024-04-22 at 17 03 38

### CODE REVIEW

**Review by rolfsmeds — APPROVED** [review]

File: src/main/java/com/vaadin/demo/component/gridpro/GridProCellEditability.java
Comment (tomivirkki) [inline src/main/java/com/vaadin/demo/component/gridpro/GridProCellEditability.java:1]: The logic is somehow wrong with the Flow example. It allows me to edit name, but not the other columns

File: src/main/java/com/vaadin/demo/component/gridpro/GridProCellEditability.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        grid.addEditColumn(Transaction::isApproved)
+                .withCellEditableProvider(
+                        transaction -> !transaction.isApproved())
+                .checkbox(Transaction::setApproved)
+                .setHeader("Approved");
+        // end::snippet[]
+        add(grid);
+    }
+
+    public static class Transaction {
```
Comment (tomivirkki) [inline src/main/java/com/vaadin/demo/component/gridpro/GridProCellEditability.java:31]: Not sure why we define a new data type for this particular example. Not a blocker.

Comment (web-padawan) [reply src/main/java/com/vaadin/demo/component/gridpro/GridProCellEditability.java]: Updated to use `Person` like other GridPro examples.

File: src/main/java/com/vaadin/demo/component/gridpro/GridProCellEditability.java
Comment (web-padawan) [reply src/main/java/com/vaadin/demo/component/gridpro/GridProCellEditability.java:1]: Replaced the example with a different one.

**Review by tomivirkki — APPROVED** [review]

**Pull request merged**